core/tauri-runtime-wry/src/lib.rs

```diff
@@ -2305,7 +2306,7 @@ impl<T: UserEvent> Runtime<T> for Wry<T> {
 pub struct EventLoopIterationContext<'a, T: UserEvent> {
   pub callback: &'a mut (dyn FnMut(RunEvent<T>) + 'static),
   pub webview_id_map: WebviewIdStore,
-  pub windows: Rc<RefCell<HashMap<WebviewId, WindowWrapper>>>,
+  pub windows: Arc<RefCell<BTreeMap<WebviewId, WindowWrapper>>>,
```

Comment:
```suggestion
  pub windows: Arc<Mutex<BTreeMap<WebviewId, WindowWrapper>>>,
```

We should use Mutex at this point even you want to use it in single thread.
This is already a control flow that has to work with tokio.
Adding lock instead of runtime check then panic is preferred behaviour imho.

Reply:
isn't RefCell better for performance? we also had some issues with deadlock before, and since using refcell (which is also a lot easier to track that kind of issue - mutating a borrowed value in this case) we've been fine